src/Conversion/ONNXToTOSA/Tensor/Flatten.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Flatten.cpp - Flatten Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX ReshapeOp to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/Interfaces/InferTypeOpInterface.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "llvm/ADT/SmallVector.h"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXFlattenLoweringToTOSA : public OpConversionPattern<ONNXFlattenOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  using OpAdaptor = typename ONNXFlattenOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXFlattenOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+
+    Value input = adaptor.input();
+    int64_t axis = adaptor.axis();
+    auto inputType = input.getType().cast<ShapedType>();
+
+    auto resultType = op.getResult().getType();
+
+    if (axis < 0) {
+      axis += inputType.getRank();
+    }
+
+    llvm::SmallVector<int64_t> newShape;
+    if (axis == 0) {
+      newShape.push_back(1);
+      int64_t lastShape = 1;
+      for (const auto &axis : inputType.getShape()) {
+        lastShape *= axis;
+      }
+      newShape.push_back(lastShape);
+    } else {
+      auto inputShape = inputType.getShape();
```

Comment:
Super nit: You can do that before the condition because you are using inputType.getShape() in the previous if.
Also, I'm not fond of `auto` if you cannot deduce the type by looking at the code and when the type is not too complex, though you can resolve this comment without changing it if you disagree.